### PR TITLE
Buffs blob laser resistance to account for the lasers buff

### PR DESCRIPTION
**Pull request by TheVekter**
## About The Pull Request
Goof said he was going to do this and then just didn't.

Regular blob tiles take half as much damage from lasers, all others take 25% less. They already have a lot of health and I think anything more than that makes it feel like a slog to kill.

## Why It's Good For The Game
Blobs feel like Swiss cheese right now if you have a laser. Enough lasers makes it completely trivial, so this should help level the playing field a little. This should _only_ impact lasers, so emitters will also get hit but other sources of heat will not.

## Changelog
:cl: Vekter
balance: Increased blob tiles' resistance to lasers to compensate for the recent buff to laser damage.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/blob/structures/factory.dm [hunk 1]
```diff
@@ -18,6 +19,11 @@
 	///Used in blob/powers.dm, checks if it's already trying to spawn a blobbernaut to prevent issues.
 	var/is_creating_blobbernaut = FALSE
 
+/datum/armor/special_factory
+	fire = 80
+	acid = 70
+	laser = 25
```
Comment (MrMelbert) [inline code/modules/antagonists/blob/structures/factory.dm:25]: Maybe subtype `/datum/armor/structure_blob` for this?

File: code/modules/antagonists/blob/structures/resource.dm [hunk 2]
```diff
@@ -6,8 +6,14 @@
 	max_integrity = BLOB_RESOURCE_MAX_HP
 	point_return = BLOB_REFUND_RESOURCE_COST
 	resistance_flags = LAVA_PROOF
+	armor_type = /datum/armor/special_resource
 	var/resource_delay = 0
 
+/datum/armor/special_resource
```
Comment (MrMelbert) [inline code/modules/antagonists/blob/structures/resource.dm:12]: Ditto

File: code/modules/antagonists/blob/structures/factory.dm (hunk 1 above)
Comment (TheVekter) [reply code/modules/antagonists/blob/structures/factory.dm:25]: This is intentional, I don't want factories and resource nodes to have the full 50 armor.

**Comment by moocowswag** [comment]
shouldnt the shield blob also get 50 or...?

also these buffs are alot stronger than the buffs lasers.

Lasers deal roughly 50% more damage from a full clip than they did before the buff you are doubling the majority of blob tile`s resistances to lasers which is why i argued for a laser armor of 30

**Comment by TheVekter** [comment]
> shouldnt the shield blob also get 50 or...?

I could have but the other blobs all felt extremely tanky if I gave them 50 so I was worried it'd be too hard to kill.

> also be mindful these buffs are alot stronger than the buffs to lasers.
> 
> Lasers deal roughly 50% more damage from a full clip than they did before the buff you are doubling the majority of blob tile`s resistances to lasers which is why i argued for a laser armor of 30 which would keep it aligned with pre buffs.

Regular blobs still die in 2 hits, I've tested this.

> Also keep in mind, a weapon that used to be printable every round and has largely only been used to fight blob is still collecting dust due to a soft removal and it might be time to make it more obtainable again in the event that this pr is either too much or we dont want lasers to be good against blob.

If we're talking about the X-Ray Laser, I don't like how it was ever _only_ good for fighting a blob and I think blob fighting is more interesting now than it used to be. It's not that we don't want lasers to be good against blobs, it's that we don't want them killing blob tiles in 1 hit.

**Comment by GuillaumePrata** [comment]
> Also keep in mind, a weapon that used to be printable every round and has largely only been used to fight blob is still collecting dust due to a soft removal and it might be time to make it more obtainable again in the event that this pr is either too much or we dont want lasers to be good against blob.

If this is about the Particle Rifle, god no, there was no counterplay the blob could do against that while it being mostly useless against anything else. 
The game is simply better without that poorly designed gun being around.

**Comment by TheVekter** [comment]
> > Also keep in mind, a weapon that used to be printable every round and has largely only been used to fight blob is still collecting dust due to a soft removal and it might be time to make it more obtainable again in the event that this pr is either too much or we dont want lasers to be good against blob.
> 
> If this is about the Particle Rifle, god no, there was no counterplay the blob could do against that while it being mostly useless against anything else. The game is simply better without that poorly designed gun being around.

Oh, the meme rifle? Yeah no, it was removed for a reason, and I'm fairly certain it wasn't "because it was good against blobs".

To elaborate on my point about x-rays, it kind of turned fighting a blob into "print five x-ray lasers and stand behind glass", which just sort of does the exact same thing lasers already do now.

**Comment by Loafin34** [comment]
> > > Also keep in mind, a weapon that used to be printable every round and has largely only been used to fight blob is still collecting dust due to a soft removal and it might be time to make it more obtainable again in the event that this pr is either too much or we dont want lasers to be good against blob.
> > 
> > 
> > If this is about the Particle Rifle, god no, there was no counterplay the blob could do against that while it being mostly useless against anything else. The game is simply better without that poorly designed gun being around.
> 
> Oh, the meme rifle? Yeah no, it was removed for a reason, and I'm fairly certain it wasn't "because it was good against blobs".
> 
> To elaborate on my point about x-rays, it kind of turned fighting a blob into "print five x-ray lasers and stand behind glass", which just sort of does the exact same thing lasers already do now.
> 
> E: I was actually wrong (shocker), part of why meme rifles were made harder to get _was_ because of blobs. Anne shifted them to kit crafting in #56133; as such you can still get them, you just have to print an upgrade kit and craft it with an energy gun, a flux anomaly core, and a gravitronic anomaly core. It's still there, just harder to get now. Way harder, actually... I should talk to a maintainer and see if we can reduce those requirements. I never see them made anymore and they're very funny.

It took hours and cooperation with the RD to get advanced eguns for two people, please fix this

**Comment by moocowswag** [comment]
thats...why...i.....said.....SOFT REMOVAL

youd be lucky to even unlock the tech after experisci let alone get 2 anomaly cores


it shoulda been nerfed if it was strong but it wasnt...people could literally just mass print them and they didnt because everyone thought it was useless.

**Comment by funman816** [comment]
please speedmerge, playing blob isnt fun anymore now that laser muskets exists (they require no recharger) and just losing progress you did in 20 minutes just gets invalidated with one assistant with a laser gun

**Comment by TheVekter** [comment]
> please speedmerge, playing blob isnt fun anymore now that laser muskets exists (they require no recharger) and just losing progress you did in 20 minutes just gets invalidated with one assistant with a laser gun

This doesn't need to be speedmerged, it's been open for over 24 hours. I'll ask someone to merge it when they have time.

**Pull request merged**